### PR TITLE
build(deps): upgrade got to 10

### PR DESCRIPTION
**Pull request by HashimotoYT**
Closes #988

### CODE REVIEW

File: script/collect.ts [hunk 1]
```diff
@@ -90,7 +90,9 @@ async function fetchApiData() {
     )
   }
 
-  const response = await got(asset.browser_download_url, { json: true })
+  const response = await got(asset.browser_download_url, {
+    responseType: 'json',
```
Comment (zeke) [inline script/collect.ts]: Good to know this API has changed. 

https://github.com/sindresorhus/got#responsetype

File: tsconfig.json [hunk 2]
```diff
@@ -9,6 +9,7 @@
     "strictNullChecks": true,
     "resolveJsonModule": true,
     "noUnusedLocals": true,
-    "noImplicitAny": true
+    "noImplicitAny": true,
+    "noEmit": true
```
Comment (zeke) [inline tsconfig.json]: This change seems unrelated. What does it do?

Comment (HashimotoYT) [reply tsconfig.json]: I run `tsc` to see types error, `noEmit` needed to don't create `.js` files and see only errors. This added in another PR.

File: script/collect.ts (hunk 1 above)
Comment (HashimotoYT) [reply script/collect.ts]: Yes, see this but I change these to `responseType: json` since `.json()` returns the only body and type `unknown`.

The `unknown` anyway returns from the `responseType` so it no makes sense to use the previous API, I change these to `json()`.
